pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/TopicsImpl.java

```diff
@@ -1231,6 +1238,48 @@ public void failed(Throwable throwable) {
         return future;
     }
 
+    @Override
+    public CompletableFuture<Void> triggerOffloadAsync(String topic, long sizeThreshold) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        try {
+            validatePersistentTopic(topic);
+            PersistentTopicInternalStats stats = getInternalStats(topic);
+            if (stats.ledgers.size() < 1) {
+                throw new PulsarAdminException("Topic doesn't have any data");
+            }
+            LinkedList<PersistentTopicInternalStats.LedgerInfo> ledgers = new LinkedList(stats.ledgers);
+            ledgers.get(ledgers.size() - 1).size = stats.currentLedgerSize; // doesn't get filled in now it seems
+            MessageId messageId = findFirstLedgerWithinThreshold(ledgers, sizeThreshold);
```

Comment:
Please check the null value. if null, we should return `CompletableFuture.completedFuture(null)`.